topo/node/srl/srl.go

```diff
@@ -37,12 +41,65 @@ func (n *Node) ConfigPush(ctx context.Context, ns string, r io.Reader) error {
 	return status.Errorf(codes.Unimplemented, "Unimplemented")
 }
 
-func (n *Node) CreateNodeResource(_ context.Context, _ node.Interface) error {
-	return status.Errorf(codes.Unimplemented, "Unimplemented")
+func (n *Node) CreateNodeResource(ctx context.Context, ni node.Interface) error {
+	log.Infof("Creating Srlinux node resource %s", n.pb.Name)
+
+	srl := srl.Srlinux{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Srlinux",
+			APIVersion: "kne.srlinux.dev/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: n.pb.Name,
+			Labels: map[string]string{
+				"app":  n.pb.Name,
+				"topo": ni.Namespace(),
+			},
+		},
+		Spec: srl.SrlinuxSpec{
+			NumInterfaces: len(ni.Interfaces()),
+			Config: &srl.NodeConfig{
+				Sleep: 0,
+			},
+		},
+	}
+	body, err := json.Marshal(srl)
+	if err != nil {
+		return err
+	}
+
+	err = ni.KubeClient().CoreV1().RESTClient().
```

Comment:
if you make a clientset to wrap this up you could refer directly to this rather than having to directly use the rest api

see https://github.com/networkop/meshnet-cni/blob/master/api/clientset/v1beta1/topology.go

for an example

Reply:
Will try, sure

Reply:
@marcushines I have a question re how to properly build a clientset rest client
for the topology rest client you have a [`New()`](https://github.com/google/kne/blob/570083e95d00fcd4f9ab5e5ceb3a1c95b63c6ee1/topo/topo.go#L63) function that tries to get the rest config from cluster first, and then resorts to `kubecfg`.

Can I do the same from the point in time where you create a custom srlinux resource? Because I don't have access to kubecfg string I think to replicate the same kind of approach

Reply:
I added for now the `*rest.Config` to node Interface here b4ec55a

Then I could use the clientset in c3ce4d5

Reply:
done